dotCMS/src/main/java/com/dotcms/cms/login/LoginServiceAPIImpl.java

```diff
@@ -0,0 +1,569 @@
+                user.setPassword(PasswordFactoryProxy.generateHash(password));
+                user.setLastLoginDate(new java.util.Date());
+                APILocator.getUserAPI().save(user, systemUser, false);
+
+                SecurityLogger.logInfo(this.getClass(),
+                                "User password was rehash with id: " + user.getUserId());
+            }
+        } catch (PasswordException | DuplicateUserException | DotDataException
+                        | DotSecurityException e) {
+            Logger.error(this.getClass(),
+                            "Error validating password from userId: " + user.getUserId(), e);
+            throw new DotSecurityException("passwords don't match");
+        }
+        return true;
+    }
+
+
+    /**
+     * Return the current login user. A {@link UserLoggingException} is thrown if any error happened
+     * getting the user.
+     *
+     * @param req
+     * @return login user
+     */
+    public User getLoggedInUser(HttpServletRequest req) {
+        User user = null;
+
+        if (req != null) {
+            try {
+                user = userWebAPI.getLoggedInUser(req);
+            } catch (PortalException | SystemException e) {
+                throw new UserLoggingException(e);
+            }
+        }
+        return user;
+    }
+
+    public User getLoggedInUser() {
+        HttpServletRequest request = this.httpServletRequestThreadLocal.getRequest();
+        return this.getLoggedInUser(request);
+    }
+
+
+    private void syncPassword(User user) {
+        try {
+
+            final boolean SYNC_PASSWORD = BaseAuthenticator.SYNC_PASSWORD;
+            if (!SYNC_PASSWORD) {
+
+                String roleName = LDAPImpl.LDAP_USER_ROLE;
+
+                if (APILocator.getRoleAPI().doesUserHaveRole(user, roleName)) {
+
+                    user.setPassword(DotCustomLoginPostAction.FAKE_PASSWORD);
+                    APILocator.getUserAPI().save(user, systemUser, false);
+                }
+            }
+        } catch (Exception e) {
+
+            Logger.debug(this.getClass(), "syncPassword not set or unable to load user", e);
+        }
+    }
+
+    private int runPreAuth(final String userName, final String password, final Company comp)
+                    throws AuthException {
+
+        if (UtilMethods.isSet(preAuthenticator) && LicenseUtil.getLevel() > 100) {
+            Authenticator authenticator =
+                            (Authenticator) ReflectionUtils.newInstance(preAuthenticator);
+
+            if (authByEmail) {
+                return authenticator.authenticateByEmailAddress(comp.getCompanyId(), userName,
+                                password);
+            } else {
+                return authenticator.authenticateByUserId(comp.getCompanyId(), userName, password);
+            }
+
+        }
+        return Authenticator.DNE;
+
+    }
+
+    private User _loadUserByKey(String key)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00100)